src/main/kotlin/com/atlassian/performance/tools/virtualusers/LoadTest.kt

```diff
@@ -110,29 +106,16 @@ internal class LoadTest(
             LinkedBlockingQueue<Runnable>(),
             ThreadFactoryBuilder().setNameFormat("virtual-user-%d").setDaemon(true).build()
         )
-        val stopSchedule = Executors.newScheduledThreadPool(
-            1,
-            ThreadFactoryBuilder().setNameFormat("deferred-stop").setDaemon(true).build()
-        )
         val segments = (1..virtualUsers).map { segmentLoad(it) }
-        val stop = stopSchedule.schedule(
-            {
-                val active = loadPool.activeCount
-                logger.info("Stopping load")
-                loadPool.shutdownNow()
```

Comment:
Here's one controversial change. I removed `shutdownNow`, because we no longer *need* to interrupt the threads. However, we still *can* do it if we feel that `done: AtomicBoolean` is not granular enough and will "waste" an entire `Action` rather than a portion of it.
IMHO it is not worth the complexity.
WDYT?